cli/src/main/java/io/quarkiverse/argocd/cli/application/GenerationBaseCommand.java

```diff
@@ -40,6 +41,13 @@ public Properties getBuildSystemProperties() {
 
     public Integer call() {
         Path projectRoot = getWorkingDirectory();
+
+        Optional<Path> scmRoot = Git.getScmRoot();
+        if (scmRoot.isEmpty()) {
+            System.out.println("Unable to determine the SCM root for the project at " + projectRoot + ". Aborting.");
+            return ExitCode.USAGE;
+        }
+
         BuildTool buildTool = QuarkusProjectHelper.detectExistingBuildTool(projectRoot);
         if (buildTool == null) {
             System.out.println("Unable to determine the build tool used for the project at " + projectRoot);
```

Comment:
"Unable to determine the build tool used for the project at" -> "Unable to determine the build automation tool used for the project at"

Reply:
The suggested change is not part of this pull request.

Reply:
Also, the quarkus code base uses the name `BuildTool` so I think we should stick to that.